resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -394,15 +476,15 @@ private boolean doResolveCached(String hostname,
 
         if (address != null) {
             setSuccess(promise, address);
-        } else if (cause != null) {
+            return true;
+        }
+        if (cause != null) {
             if (!promise.tryFailure(cause)) {
```

Comment:
should we make this a function like `setSuccess` and use it where we call `tryFailure` in `doResolveAllCached`?